TrafficCapture/trafficReplayer/src/test/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonToByteBufHandlerTest.java

```diff
@@ -77,22 +56,35 @@ public void testWithSeed(Random random) {
                 .map(i->writeAndCheck(channel, preSizesFromUpstreamHandler, genRandom.get()))
                 .sum();
         channel.close();
-        Assertions.assertEquals(totalBytesWritten, validationHandler.totalBytesRead);
-        var postSizes = validationHandler.chunkSizesReceivedList;
-        var commonLength = Math.min(asCapturedSizes.size(), postSizes.size()) - 1;
+
+        var chunkSizesReceivedList = getByteBufSizesFromChannel(channel);
+        Assertions.assertEquals(totalBytesWritten, chunkSizesReceivedList.stream().mapToInt(x->x).sum());
+        var commonLength = Math.min(asCapturedSizes.size(), chunkSizesReceivedList.size()) - 1;
         Assertions.assertArrayEquals(
                 asCapturedSizes.subList(0, commonLength).toArray(),
-                postSizes.subList(0, commonLength).toArray());
+                chunkSizesReceivedList.subList(0, commonLength).toArray());
         var postSizesPastSourceConstrained =
-                commonLength+1 < postSizes.size() ? postSizes.subList(commonLength+2, postSizes.size()) : List.of();
+                commonLength+1 < chunkSizesReceivedList.size() ?
+                        chunkSizesReceivedList.subList(commonLength+2, chunkSizesReceivedList.size())
+                        : List.of();
         Assertions.assertArrayEquals(preSizesFromUpstreamHandler.subList(preSizesFromUpstreamHandler.size()-postSizesPastSourceConstrained.size(),
                         preSizesFromUpstreamHandler.size()).toArray(),
                 postSizesPastSourceConstrained.toArray());
 
-        Assertions.assertTrue(validationHandler.chunkSizesReceivedList.size() >
+        Assertions.assertTrue(chunkSizesReceivedList.size() >
                 sharedInProgressChunkSizes.size()+1);
     }
 
+    private static List<Integer> getByteBufSizesFromChannel(EmbeddedChannel channel) {
+        return channel.inboundMessages().stream()
+                .map(x -> {
+                    var bb = ((ByteBuf) x);
+                    bb.release();
+                    return bb.readableBytes();
+                })
+                .collect(Collectors.toList());
+    }
```

Comment:
Can the sizes here be collected and then have the bytebufs released? I'm concerned about how we're returning `bb.readableBytes` after `bb.release()` is called. Is that valid?

Reply:
I think that it is today, but it's at the very least, gross (and not future proof).  I'll fix this with a finally block or something like it..

Reply:
Fixed in the cleanup commit